### PR TITLE
ARROW-13213: [C++][Gandiva] Implement parse_url hive function

### PR DESCRIPTION
**Pull request by jvictorhuguenin**

### CODE REVIEW

**Comment by jvictorhuguenin** [comment]
@jpedroantunes, can you review this PR, please?

File: cpp/src/gandiva/precompiled/string_ops.cc [hunk 1]
```diff
@@ -1966,4 +1966,233 @@ const char* binary_string(gdv_int64 context, const char* text, gdv_int32 text_le
...
+
+FORCE_INLINE
+const char* parse_url_utf8_utf8(gdv_int64 context, const char* url, gdv_int64 url_len,
+                                const char* part_to_extract, gdv_int64 part_len,
+                                gdv_int32* out_len) {
+  if (url_len == 0) {
+    return nullptr;
+  }
+
+  auto protocol_key = "PROTOCOL";
```
Comment (jpedroantunes) [inline cpp/src/gandiva/precompiled/string_ops.cc:1978]: Maybe we can check a way of not having to instantiate these variables constantly.

File: cpp/src/gandiva/precompiled/string_ops_test.cc [hunk 2]
```diff
@@ -1540,4 +1540,72 @@ TEST(TestStringOps, TestConvertToBigEndian) {
   }
 #endif
 }
+
```
Comment (jpedroantunes) [inline cpp/src/gandiva/precompiled/string_ops_test.cc:1543]: Add prijector tests please

**Review by jpedroantunes — CHANGES_REQUESTED** [review]

**Comment by amol-** [comment]
Closing because it has been untouched for a while, in case it's still relevant feel free to reopen and move it forward 👍

**Pull request closed without merging**